src/EditorFeatures/CSharpTest/Completion/CompletionProviders/KeywordCompletionProviderTests.cs

```diff
@@ -317,5 +317,287 @@ class C
 }";
             await VerifyProviderCommitAsync(markupBeforeCommit, "return", expectedCodeAfterCommit, commitChar: ';', textTypedSoFar: "return");
         }
+
+        [Test.Utilities.CompilerTrait(Test.Utilities.CompilerFeature.ReadonlyReferences)]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task ReadOnlyWillPopUpAfterRefInMethodParameterModifiers_Methods()
```

Comment:
wrong place for all these tests.  Tests for specific keywords should be in things like:

ReadOnlyKeywordRecommenderTests
RefKeywordRecommenderTests
InKeywordRecommenderTests

these tests will validate that everything works properly (including bringing up completion at the end of the file, or if the keyword is partially written).